File: src/main/java/phoenix/utils/ArrayUtils.java
package phoenix.utils;

import java.util.ArrayList;
import java.util.Arrays;

public class ArrayUtils
{
    static public <T extends Object> ArrayList<T> sumArrays(ArrayList<T> first, ArrayList<T> second, T... adv)
    {
        ArrayList<T> res = new ArrayList(first);
        res.addAll(second);

        res.addAll(Arrays.asList(adv));
        return res;
    }

    @SafeVarargs
    static public <T extends Object> ArrayList<T> sumArrays(ArrayList<T> first, T... adv)
    {
        ArrayList<T> res = new ArrayList<>(first);
        res.addAll(Arrays.asList(adv));
        return res;
    }
}

File: src/main/kotlin/phoenix/utils/ArrayUtils.kt
package phoenix.utils

import java.util.*


object ArrayUtils
{
    fun <T> part(list: ArrayList<T>, from: Int, to: Int): ArrayList<T>
    {
        val res = ArrayList<T>()
        for (i in from until to)
        {
            res.add(list[getIndex(list.size, i)])
        }
        return res
    }

    fun <T> part(list: LinkedList<T>, from: Int, to: Int): ArrayList<T>
    {
        val res = ArrayList<T>()
        var i = from
        while (i < to && i < list.size)
        {
            res.add(list[getIndex(list.size, i)])
            i++
        }
        return res
    }

    private fun getIndex(size: Int, indexIn: Int): Int
    {
        var index = indexIn
        if (size == 0) return 0
        while (index < 0) index += size
        return if (index >= size) size - 1 else index
    }
}
